submissions/iva-stasia/frogger-game/app.js

```diff
@@ -0,0 +1,184 @@
+        <img src='images/char-boy.png' class='player_img' data-player-num = '0' />
+        <img src='images/char-cat-girl.png' class='player_img' data-player-num = '1' />
+        <img src='images/char-princess-girl.png' class='player_img' data-player-num = '2' />
+    `;
+
+/*---------- Enemy ----------*/
+
+const Enemy = function(x, y, speed) {
+    this.x = x;
+    this.y = y;
+    this.speed = speed;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.update = function(dt) {
+    this.x += this.speed * dt + levelSpeed;
+
+    if (this.x > INIT_VAL.canvasWidth
+        || !playerOptionDiv.classList.contains('hide')
+        || lostMessage.classList.contains('active_message')
+        || winMessage.classList.contains('active_message')) {
+        this.x = INIT_VAL.enemyX;
+    };
+
+    this.checkCollision();
+};
+
+Enemy.prototype.checkCollision = function() {
+    if (player.x < this.x + INIT_VAL.collision &&
+        player.x + INIT_VAL.collision > this.x &&
+        player.y < this.y + INIT_VAL.collision &&
+        INIT_VAL.collision + player.y > this.y) {
+            player.resetGame();
+    };
+};
+
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const allEnemies = [];
+
+INIT_VAL.enemyY.forEach(y => {
+    let enemySpeed = Math.round(Math.random() * INIT_VAL.enemyMaxspeed + INIT_VAL.enemyMinSpeed);
+    let enemy = new Enemy(INIT_VAL.enemyX, y, enemySpeed);
+    allEnemies.push(enemy);
+});
+
+/*---------- Player ----------*/
+
+const Player = function(x, y) {
+    this.x = x;
+    this.y = y;
+    this.sprite = 'images/char-boy.png';
+
+    playerOptionDiv.addEventListener('click', (event) => {
+        const playerNum = +event.target.dataset.playerNum;
+        this.sprite = allPlayerSprites[playerNum];
+        playerOptionDiv.classList.add('hide');
+    });
+};
+
+Player.prototype.update = function() {
+    
+};
+
+Player.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Player.prototype.resetGame = function() {
+    allEnemies.forEach(enemy => enemy.x = INIT_VAL.enemyX);
+    this.x = INIT_VAL.playerX;
+    this.y = INIT_VAL.playerY;
+    levelSpeed = 0;
+    level = 0;
+    levelDiv.innerHTML = `<div class='level'>Level: ${level}/5</div>`;
+    lostMessage.classList.add('active_message');
+};
+
+Player.prototype.upLevel = function() {
+    this.x = INIT_VAL.playerX;
+    this.y = INIT_VAL.playerY;
+    levelSpeed++;
+    level++;
+    levelDiv.innerHTML = `<div class='level'>Level: ${level}/5</div>`;
+
+    if (level === 5) {
+        winMessage.classList.add('active_message');
+    };
+};
+
+const player = new Player(INIT_VAL.playerX, INIT_VAL.playerY);
+
+Player.prototype.handleInput = function(allowedKeys) {
+    if (!lostMessage.classList.contains('active_message')
+        && !winMessage.classList.contains('active_message')
+        && playerOptionDiv.classList.contains('hide')) {
```

Comment:
Making game objects dependant on environment is not a good practice.
The better practice would be to provide class with some external functions of interacting with the excution environment.
You do not need to fix this in this task.
Keep this principle in mind.

Reply:
Thank you for this note.